spotbugs/src/main/java/edu/umd/cs/findbugs/ba/vna/ValueNumberFrame.java

```diff
@@ -578,31 +590,76 @@ public boolean haveMatchingFlags(ValueNumber v1, ValueNumber v2) {
     }
 
     public Collection<ValueNumber> valueNumbersForLoads() {
-        HashSet<ValueNumber> result = new HashSet<>();
         if (REDUNDANT_LOAD_ELIMINATION) {
-            for (Map.Entry<AvailableLoad, ValueNumber[]> e : getAvailableLoadMap().entrySet()) {
-                if (e.getValue() != null) {
-                    Collections.addAll(result, e.getValue());
-                }
-            }
+            return Collections.unmodifiableCollection(getAvailableLoadReverseMap().keySet());
         }
 
-        return result;
+        return Collections.emptySet();
```

Comment:
Not sure how existing callers expect, but if there is a caller expecting mutability, it could be better to return a `new HashSet<>()`.